Makefile

```diff
@@ -93,27 +113,38 @@ binary: $(CALICOCTL_FILES) vendor
 	fi; \
 	GOOS=$(OS) GOARCH=$(ARCH) CGO_ENABLED=0 go build -v $$INSTALL_FLAG -o dist/calicoctl-$(OS)-$(ARCH) $(LDFLAGS) "./calicoctl/calicoctl.go"
 
+
 dist/calicoctl: $(CALICOCTL_FILES) vendor
-	$(MAKE) dist/calicoctl-linux-$(ARCH)
-	cp dist/calicoctl-linux-$(ARCH) dist/calicoctl
+	$(MAKE) dist/calicoctl-linux-amd64
+	cp dist/calicoctl-linux-amd64 dist/calicoctl
+
+## build for linux on amd64
+dist/calicoctl-linux-amd64:
```

Comment:
These per-arch targets need the dependencies too.  I think the confusion here is that you're using these as "command" targets (like `make clean`), but the problem is that they actually do create the target file `dist/calicoctl-linux-amd64`.  If that file happens to exist, make will decide that it's never stale (because this target has no dependencies).

The same would happen with `make clean` if you created a file called `clean`.

There are two ways around it: 

- add the dependencies back
- mark the targets as phony so that Make knows to always rebuild them with `.PHONY: dist/calicoctl-linux-amd64`

Reply:
Yes, good point. This is why we have code review! I am putting them back now and squashing.